### PR TITLE
[WIP] Fixes GH243 Added dashboards pluggin

### PR DESCRIPTION
**Pull request by lxatrhea**
**WIP**: Unit tests

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [ ] I have provided test coverage for my change (where applicable)

### Description
- Dashboard plugin is completely new
- Added CDP4Colors enumeration for dashboard backcolor
- Refactored colors in Requirement Verification, so it uses CDP4Color
- Installer has been updated

### CODE REVIEW

File: CDP4Composition/Utilities/CDP4ColorExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿// -----------------------------------------------------------------------------------------------
+// <copyright file="CDP4ColorExtensions.cs" company="RHEA System S.A.">
+//   Copyright (c) 2020 RHEA System S.A.
+// </copyright>
+// -----------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.Utilities
```
Comment (samatrhea) [inline CDP4Composition/Utilities/CDP4ColorExtensions.cs]: please consider adding unit tests

File: CDP4Composition/Utilities/Cdp4Color.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+// -----------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.Utilities
+{
+    /// <summary>
+    /// Enum used to define default CDP4 system colors
+    /// </summary>
+    public enum CDP4Color
+    {
+        Failed,
```
Comment (samatrhea) [inline CDP4Composition/Utilities/Cdp4Color.cs]: can you add some docs to each enumeration value?

File: CDP4Composition/Utilities/RequirementStateOfComplianceExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+namespace CDP4Composition.Utilities
+{
+    using System.Windows.Media;
+
+    using CDP4RequirementsVerification;
+
+    /// <summary>
+    /// Extension methods for the <see cref="RequirementStateOfCompliance"/> enum
+    /// </summary>
+    public static class RequirementStateOfComplianceExtensions
```
Comment (samatrhea) [inline CDP4Composition/Utilities/RequirementStateOfComplianceExtensions.cs]: please consider adding unit tests

File: CDP4Dashboard/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (samatrhea) [inline CDP4Dashboard/Properties/AssemblyInfo.cs]: this is not yet a version one, please update to 0.01.0

File: CDP4Dashboard/Views/Widget/SizeObserver.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+// <copyright file="SizeObserver.cs" company="RHEA System S.A.">
+// This is a Code Snippet found here: https://gist.github.com/jrgcubano/54ecbc61cbfaaa83b4dd14de03adf296
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Dashboard.Views.Widget
+{
+    using System.Windows;
+
+    /// <summary>
```
Comment (samatrhea) [inline CDP4Dashboard/Views/Widget/SizeObserver.cs]: missing docs on members

File: CDP4IMEInstaller/CDP4Dashboard.wxs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  <Fragment>
+    <ComponentGroup Id="CDP4DASHBOARD_CG" Directory="DASHBOARD">
+      
+      <Component Id="comp_CDP4Dashboard" Guid="{60DCB213-9D8E-4C5F-984F-94755EDB0397}" Location="local" Win64="$(var.Win64)">
+        <File Id="fil_CDP4Dashboard"  Vital="yes" Source="$(var.CDP4Dashboard.TargetPath)" KeyPath="yes"></File>
+      </Component>
+
+      <Component Id="comp_chartscore" Guid="{DDC8052B-365B-41D8-98CB-BCF0A2BC4EB8}" Location="local" Win64="$(var.Win64)">
+        <File Id="fil_chartscore" Vital="yes" Source="$(var.CDP4Dashboard.TargetDir)DevExpress.Charts.v19.2.Core.dll" KeyPath="yes" />
```
Comment (samatrhea) [inline CDP4IMEInstaller/CDP4Dashboard.wxs]: any references to devexpress components need to be in the following file `CDP4IME.wxs` and not in the plugin specific parts.

**Comment by lxatrhea** [comment]
> 
> 
> code looks good and i also played around with it a bit, very cool stuff
> 
> I do have concern that there is no test project and that none of the code is developed using a unit testing approach. I am not saying that it you need to do pure TDD, but how can you make sure it works the way you want without it, unless you continously compile and run the app, which in my opinion takes too much time...

I understand. Unit tests are not finished, that's why there is a WIP indication.
But indeed, I didn't do any TDD here. I did some for the related SDK and Webservice changes but not for the IME.

File: CDP4Dashboard/ViewModels/Widget/Base/WidgetDetailsBase.cs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        public IDialogResult DialogResult
+        {
+            get => this.dialogResult;
+            set => this.RaiseAndSetIfChanged(ref this.dialogResult, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the busy indicator
+        /// </summary>
+        public bool IsBusy { get; set; }
```
Comment (alexatrhea) [inline CDP4Dashboard/ViewModels/Widget/Base/WidgetDetailsBase.cs]: Are these still needed? if yes, then they should be reactive?

**Comment by lxatrhea** [comment]
Unit Test project will be added in another issue.

**Pull request merged**